src/test/java/seedu/intrack/logic/commands/AddTagCommandTest.java

```diff
@@ -0,0 +1,98 @@
+package seedu.intrack.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.intrack.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.intrack.logic.commands.CommandTestUtil.showInternshipAtIndex;
+import static seedu.intrack.testutil.TypicalIndexes.INDEX_FIRST_INTERNSHIP;
+import static seedu.intrack.testutil.TypicalIndexes.INDEX_SECOND_INTERNSHIP;
+import static seedu.intrack.testutil.TypicalInternships.getTypicalInTrack;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.intrack.commons.core.Messages;
+import seedu.intrack.commons.core.index.Index;
+import seedu.intrack.model.Model;
+import seedu.intrack.model.ModelManager;
+import seedu.intrack.model.UserPrefs;
+import seedu.intrack.model.tag.Tag;
+
+
+class AddTagCommandTest {
+
+    private Model model = new ModelManager(getTypicalInTrack(), new UserPrefs());
+
+    private Tag urgent = new Tag("Urgent");
+    private Tag remote = new Tag("Remote");
+    private Tag help = new Tag("Help");
+
+    private List<Tag> taglistStub = new ArrayList<Tag>(Arrays.asList(remote,
+            urgent, help));
+
+    private List<Tag> taglistStub1 = new ArrayList<Tag>(Arrays.asList(urgent, help, remote));
+
+    private Set<Tag> taglistStub2 = new HashSet<Tag>(Arrays.asList(urgent, help, remote));
+
+
+    /* I am not sure why this is causing the toModelType_typicalInternshipsFile_success()  to fail in json tests
```

Comment:
Have you tried using the debugger to find out why? Might be able to see which fields are different and causing the mismatch

Reply:
Thank you for the suggestion, I've got it to work!